### PR TITLE
[233] Signup/login/logout with google

### LINKED ISSUE #233: User Sign up/Log in (Manager, Reader)
As a user I want to be able to sign up the system so that I can use the library app

**Acceptance criteria**
1. Google sign up
2. Manager role will by preexisting account

### PR DESCRIPTION
**Pull request by x8er**
image
closes #233

### CODE REVIEW

File: packages/apollo-client/src/lib/links/refreshTokenLink.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  const currentGMTTimestamp =
+    Date.now() - Math.abs(new Date(Date.now()).getTimezoneOffset()) * 60000;
+
+  if (currentGMTTimestamp > parseInt(expiry_date)) {
+    const { data } = await axios.post<{
+      access_token: string;
+      id_token: string;
+      expiry_date: string;
+    }>('http://localhost:3333/api/auth/refresh-token', {
```
Comment (sabrina34352) [inline packages/apollo-client/src/lib/links/refreshTokenLink.ts:18]: i think you should discuss with Ruslan which URL should be on prod, or take url from env file

**Review by sabrina34352 — CHANGES_REQUESTED** [review]

File: packages/apollo-client/src/lib/links/refreshTokenLink.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      refresh_token,
+    });
+
+    console.log(data);
```
Comment (sabrina34352) [inline packages/apollo-client/src/lib/links/refreshTokenLink.ts:22]: console.log shouldnt be there

File: packages/frontapp/src/app/pages/StartPage.tsx [hunk 3]
```diff
@@ -52,137 +74,101 @@ const StartPageParagraph = styled.p`
...
-    dispatch(setUser({ username, location }));
+  const [updatePersonLocationMutation] = useUpdatePersonLocationMutation();
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+  const history = useNavigate();
+  const googleLogin = useGoogleLogin({
+    flow: 'auth-code',
+    onSuccess: async ({ code }) => {
+      const { data } = await axios.post<TAuthResponseData>(
+        'http://localhost:3333/api/auth',
```
Comment (sabrina34352) [inline packages/frontapp/src/app/pages/StartPage.tsx:99]: here url too

**Review by sabrina34352 — CHANGES_REQUESTED** [review]

**Review by unrealsolver — APPROVED** [review]

**Review by IvanUglovets — APPROVED** [review]

File: packages/apiserver/src/resources/persons/person.resolver.ts [hunk 4]
```diff
@@ -6,27 +6,38 @@ import {
...
 @Resolver('Person')
 export class PersonResolver {
   @Query(() => [Person])
+  @UseGuards(AuthGuard)
   async getAllPersons() {
     return Person.find();
   }
 
   @Query(() => Person)
+  @UseGuards(AuthGuard)
```
[the remark this answers is not in the document]
Comment (x8er) [reply packages/apiserver/src/resources/persons/person.resolver.ts:34]: No for all endpoints

**Pull request merged**